.github/actions/gem-dependency-updater/scripts/gem_dependency_updater.rb

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env ruby
+
+require 'json'
+
+class GemDependencyUpdater
+  USAGE_MESSAGE = "Usage: #{$0} <branch_name> <gem_name>"
+  GIT_USER_NAME = 'sofatutor-bot'
+  GIT_USER_EMAIL = 'operations+github-bot@sofatutor.com'
+  GEMFILE_PATH = 'Gemfile'
+  BASE_BRANCH = 'main'
+
+  def initialize(branch_name:, gem_name:)
+    @github_event = JSON.parse(ENV['GITHUB_EVENT'])
+
+    @branch_name = branch_name
+    @gem_name = gem_name
+
+    @dependent_repo_branch_name =
+      if is_merge_event?
+        @github_event['pull_request']['head']['ref']
+      else
+        @branch_name
+      end
+
+    validate_arguments
+  end
+
+  def call
+    checkout_branch
+    update_gem_dependency
+    commit_and_push_changes
+    create_pull_request
+  end
+
+  private
+
+  def is_merge_event?
+    return false unless @github_event['action'] == 'closed'
+    return false unless @github_event['pull_request']
+
+    @github_event['pull_request']['merged'] == true
+  end
+
+  def validate_arguments
+    abort(USAGE_MESSAGE) if @branch_name.nil? || @branch_name.strip.empty? || @gem_name.nil? || @gem_name.strip.empty?
+  end
+
+  def checkout_branch
+    execute_command("git fetch --depth=1 origin", "Failed to fetch from origin.")
```

Comment:
Oh I didn't know this `--depth=1` interesting! Are you doing this just to be safe?